LabExT/View/MeasurementControlSettings.py

```diff
@@ -0,0 +1,220 @@
+
+
+class MeasurementControlSettingsView:
+    """ Modify how measurement storing and tracking are handled """
+
+    def __init__(self, parent, experiment_manager):
+        self._root = parent
+        self.exp_manager = experiment_manager
+        self.logger = logging.getLogger()
+
+        self._settings = MeasurementControlSettings()
+
+        self.settings_frame = None
+        self.limit_finished_meas_button = None
+
+        self.measurements_limited = BooleanVar(self._root, value=self._settings.finished_meas_limited)
+        self.measurements_limited.trace("w", self.measurements_checkbox_changed)
+        self.measurement_limit = StringVar(self._root, value=str(self._settings.max_finished_meas))
+        self.measurement_limit_label = None
+        self.measurement_limit_field = None
+
+        self.todos_limited = BooleanVar(self._root, value=self._settings.displayed_todo_limited)
+        self.todos_limited.trace("w", self.todos_checkbox_changed)
+        self.todo_limit = StringVar(self._root, value=str(self._settings.max_displayed_todo))
+        self.todo_limit_label = None
+        self.todo_limit_field = None
+
+        self.no_json_indentation = BooleanVar(self._root, value=not self._settings.json_indented)
+
+        # draw GUI
+        self.__setup__()
+
+    def measurements_checkbox_changed(self, *args) -> None:
+        if self.measurements_limited.get():
+            self.measurement_limit_label.config(state="normal")
+            self.measurement_limit_field.config(state="normal")
+        else:
+            self.measurement_limit_label.config(state="disabled")
+            self.measurement_limit_field.config(state="disabled")
+
+    def todos_checkbox_changed(self, *args) -> None:
+        if self.todos_limited.get():
+            self.todo_limit_label.config(state="normal")
+            self.todo_limit_field.config(state="normal")
+        else:
+            self.todo_limit_label.config(state="disabled")
+            self.todo_limit_field.config(state="disabled")
+
+    def _validate_entries(self):
+        max_meas = int(self.measurement_limit.get())
+        if max_meas <= 0:
+            raise ValueError(f'The maximum number of measurement displayed cannot be lower than 1. Got {max_meas}')
+        max_todos = int(self.todo_limit.get())
+        if max_todos <= 1:
+            raise ValueError(f'The maximum number of ToDos displayed cannot be lower than 1. Got {max_todos}')
+
+    def save_and_close(self) -> None:
+        self._validate_entries()
+        self._settings.finished_meas_limited = self.measurements_limited.get()
+        self._settings.max_finished_meas = int(self.measurement_limit.get())
+        self._settings.displayed_todo_limited = self.todos_limited.get()
+        self._settings.max_displayed_todo = int(self.todo_limit.get())
+        self._settings.json_indented = not self.no_json_indentation.get()
+
+        self._settings.save_to_file()
+        self.window.destroy()
```

Comment:
I suggest some functionality here to ask the main window to redraw the todo table here to immediately reflect any changes in the settings that the user made.

I'm pretty sure we used the "reload todo tables" functionality exists somewhere and can be reused.

Reply:
good call! We now call the `update_tables()` of the `MainWindowController` on the save_and_close call. This will update the ToDo table based on the new settings immediately. For the measurement table however, this does currently not work, as the adding and removing of measurements is handled in the `StandardExperiment`. For this to work, we would need to separate the loaded meausrements from the `StandardExperiment` and this would require a bit more work.